test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using ActionResultsWebSite;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    public class ActionResultTests
+    {
+        private readonly IServiceProvider _provider = TestHelper.CreateServices("ActionResultsWebSite");
+        private readonly Action<IApplicationBuilder> _app = new Startup().Configure;
+
+        [Fact]
+        public async Task BadRequestResult_CanBeReturned()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            var input = "{\"SampleInt\":10}";
+
+            var request = new HttpRequestMessage(HttpMethod.Post, "http://localhost/Home/GetBadResult");
+            request.Content = new StringContent(input, Encoding.UTF8, "application/json");
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            Assert.Equal("", await response.Content.ReadAsStringAsync());
+        }
+
+        [Fact]
+        public async Task CreatedResult_Relative()
```

Comment:
Could you come up with a better name for this test and the one below? It's not entirely obvious what it's attempting to test from the name.